src/quorum/datadriven_test.rs

```diff
@@ -0,0 +1,142 @@
+use crate::quorum::{AckIndexer, Index};
+use crate::{default_logger, HashSet, JointConfig, MajorityConfig};
+use datadriven::{run_test, TestData};
+
+#[test]
+fn test_data_driven_quorum() -> anyhow::Result<()> {
+    let logger = default_logger();
+    fn test_quorum(data: &TestData) -> String {
```

Comment:
Better move it out of function for readability.

Reply:
resolved.